src/cpp/include/configuration/config.h

```diff
@@ -152,6 +152,7 @@ struct TrainingConfig {
     shared_ptr<CheckpointConfig> checkpoint = nullptr;
```

Comment:
Is this checkpoint config used anywhere?

My original purpose was to have this as nested option for the checkpointing stuff. E.g.

```
training:
  ...
  checkpoint:
    interval: 10
```

So either this should be deleted or the other checkpoint options should be merged into it. What do you think?